streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -554,6 +563,10 @@ private void runLoop() {
                                  "trigger a new rebalance.");
                     assignmentErrorCode.set(AssignorError.NONE.code());
                     mainConsumer.enforceRebalance();
+                } else if (nextProbingRebalanceMs.get() < time.milliseconds()) {
+                    log.info("The probing rebalance interval has elapsed since the last rebalance, triggering a " +
+                                "rebalance to probe for newly caught-up clients");
+                    mainConsumer.enforceRebalance();
```

Comment:
req: I guess you could also verify this with a mock consumer.

Reply:
added `shouldEnforceRebalanceAfterNextScheduledProbingRebalanceTime`